### DISCUSSION TITLE
Server optimisation of large grous vs. small groups

### DISCUSSION THREAD

**Discussion by pianojoe** [jamulussoftware/jamulus#1205] [Technical Support & Help] [answered]
I’m running choir rehearsals with 40–50 singers on a virtual private server in the cloud, 4 vCPUs. I understand that Jamulus was originally meant for small bands, maybe 10 musicians. I’d rather have more people join than to squeeze the last millisecond of latency out of my setup. 

I have two questions:

(1) --fastupdate  increases server load and bandwidth needs, so should it be left off in my case?

(2) Will a low-latency kernel on the server (Ubuntu 20.04 LTS) help, or does this usually affect local audio performance only? ("If a Jamulus server runs in a data center, and no-one is there to listen, will it make a sound?" 😉)

**Comment by dcorson-ticino-com** [comment, marked as answer]
Hi Joe,
The most important is that you use -T to enable multi-threading.
I would not disable fastupdate as for those that use 64 bit small buffers on their clients that makes a real difference of a several ms in the latency, but that is a possibility if you are really pushing the limits.
As I understand it 20.04 uses the low latency kernel by default.
I would not worry too much, I run my server on an Odroid-N2 which is a better raspPI 4, maybe twice as fast, and have had rehearsals of 25 people running without a hitch.  According to the tests I have done I think it would be OK to about 35 clients, but I have set the max to 25 just to have margin.  If you are talking of Intel cores you should have power to spare.  The server has an amazingly small footprint for all that it does.
Try doing a top while everyone is logged in to see how much processor power you are actually using and how well the load is spread over the cores.
Don

**Comment by ann0see** [comment]
Just marked @dcorson-ticino-com 's comment as answer. I hope this is ok for you?
